File: frontend/src/types/health.ts
export interface HealthResponse {
    status: string;
    service: string;
}

export interface DatabaseHealthResponse {
    status: string;
    database: string;
    detail?: string | null;
}

export interface StorageHealthResponse {
    status: string;
    storage: string;
    bucket: string;
    detail?: string | null;
}

export interface FullHealthResponse {
    status: string;
    backend: string;
    database: string;
    storage: string;
}

File: backend/app/schemas/health.py
from typing import Optional

from pydantic import BaseModel


class HealthResponse(BaseModel):
    status: str
    service: str


class DatabaseHealthResponse(BaseModel):
    status: str
    database: str
    detail: Optional[str] = None


class StorageHealthResponse(BaseModel):
    status: str
    storage: str
    bucket: str
    detail: Optional[str] = None


class FullHealthResponse(BaseModel):
    status: str
    backend: str
    database: str
    storage: str
